verification/tests/solidity_multiple_types.rs

```diff
@@ -0,0 +1,85 @@
+#![allow(dead_code, unused)]
+
+use std::collections::BTreeMap;
+
+pub struct TestInput {
+    pub contract_name: &'static str,
+    pub compiler_version: &'static str,
+    pub evm_version: &'static str,
+    pub optimization_runs: Option<usize>,
+    pub contract_libraries: BTreeMap<String, String>,
+    pub has_constructor_args: bool,
+
+    /// If None, the input would be read from the corresponding file
+    pub source_code: Option<String>,
+    /// If None, the input would be read from the corresponding file
+    pub creation_tx_input: Option<String>,
+    /// If None, the bytecode would be read from the corresponding file
+    pub deployed_bytecode: Option<String>,
+}
+
+impl TestInput {
+    pub fn new(contract_name: &'static str, compiler_version: &'static str) -> Self {
+        Self {
+            contract_name,
+            compiler_version,
+            evm_version: "default",
+            optimization_runs: None,
+            contract_libraries: Default::default(),
+            has_constructor_args: false,
+
+            source_code: None,
+            creation_tx_input: None,
+            deployed_bytecode: None,
+        }
+    }
+
+    pub fn with_evm_version(self, evm_version: &'static str) -> Self {
```

Comment:
I understand that this is a code for test, and it can be not fully optimized, but still, passing self by reference removes all the copying/moving, while this variant still moves each variable
https://godbolt.org/z/1oqzjhfWj
https://godbolt.org/z/c7h6WeKEj
So I suggest changing this to reference (everywhere)

Also I really don't understand why
```rust
let input = TestInput::new(..).with_evm_version(something);
```
is better, than
```rust
let input = TestInput::new(..);
input.evm_version = something;
```
or
```rust
let input = TestInput{
  evm_version,
  optimization_runs: 200,
  ..TestInput::new(..),
}
```

Reply:
Passing self by value allows us to create a chain of calls which initiates the structure instead of making a separate statement for each field. Likewise, using setters on construction allows us to hide some implementation details not important for the initialization (e.g. we pass a `String` as a deployed_bytecode in `TestInput::with_deployed_bytecode`, while internally the method sets corresponding field to be `Some(deployed_bytecode)`. In my opinion, altogether they make the initialization logic more explicit and readable. 

Regarding optimization I believe clearness is more preferable here in comparison with performance gain which we get by using references. However, I've updated the code so that it has only one more move operation in comparison with using references while leaving the user experience the same (https://godbolt.org/z/MMnn8bTzq)